tests/ui_tests/test_jobs_page.py

```diff
@@ -0,0 +1,197 @@
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Union
+
+import os
+import pytest
+import allure
+
+from adcm_client.objects import ADCMClient, Cluster, Bundle, Provider, Service, Host
+from adcm_pytest_plugin import utils
+from adcm_pytest_plugin.steps.actions import run_cluster_action_and_assert_result
+
+from tests.ui_tests.app.app import ADCMTest
+from tests.ui_tests.app.page.job_list.page import (
+    JobListPage,
+    JobStatus,
+    TableTaskInfo,
+    TaskInfo,
+)
+
+
+LONG_ACTION_DISPLAY_NAME = 'Long action'
+SUCCESS_ACTION_DISPLAY_NAME = 'Success action'
+FAIL_ACTION_DISPLAY_NAME = 'Fail action'
+
+CLUSTER_NAME = 'Great Cluster'
+SERVICE_NAME = 'Serviceno'
+
+# pylint: disable=redefined-outer-name
+
+
+@pytest.fixture()
+def page(app_fs: ADCMTest, login_to_adcm_over_api) -> JobListPage:
+    return JobListPage(app_fs.driver, app_fs.adcm.url).open()
+
+
+@allure.title("Upload cluster bundle")
+@pytest.fixture()
+def cluster_bundle(sdk_client_fs: ADCMClient) -> Bundle:
+    return sdk_client_fs.upload_from_fs(os.path.join(utils.get_data_dir(__file__), "cluster"))
+
+
+@allure.title("Upload provider bundle")
+@pytest.fixture()
+def provider_bundle(sdk_client_fs: ADCMClient) -> Bundle:
+    return sdk_client_fs.upload_from_fs(os.path.join(utils.get_data_dir(__file__), "provider"))
+
+
+@allure.title("Create cluster")
+@pytest.fixture()
+def cluster(cluster_bundle: Bundle) -> Cluster:
+    """Creates cluster and adds service"""
+    cluster = cluster_bundle.cluster_create(CLUSTER_NAME)
+    cluster.service_add(name=SERVICE_NAME)
+    return cluster
+
+
+@allure.title("Create provider")
+@pytest.fixture()
+def provider(provider_bundle: Bundle) -> Provider:
+    return provider_bundle.provider_create('Awesome Provider')
+
+
+def test_cluster_action_job(cluster: Cluster, page: JobListPage):
+    """Run action on cluster and validate job in table and popup"""
+    _test_run_action(page, cluster)
+
+
+def test_service_action_job(cluster: Cluster, page: JobListPage):
+    """Run action on service and validate job in table and popup"""
+    _test_run_action(page, cluster.service_list()[0])
+
+
+def test_provider_action_job(provider: Provider, page: JobListPage):
+    """Run action on host provider and validate job in table and popup"""
+    _test_run_action(page, provider)
+
+
+def test_host_action_job(provider: Provider, page: JobListPage):
+    """Run action on host and validate job in table and popup"""
+    _test_run_action(page, provider.host_create('some-fqdn'))
+
+
+def test_run_successful_job(cluster: Cluster, page: JobListPage):
```

Comment:
looks like test_run_successful_job and test_run_fail_job are quite similar, mb parametrize?

Reply:
Tried to, guess it looks okay now